vtex/node-vtex-api
# Fix node cluster workers heap size

#### What is the purpose of this pull request?
We started using `cluster` in the node runtime but forgot to
tune the forked processes heap sizes so that they wouldn't
each of them try to use all the memory that was available
in the container. This is measurable in our most memory-intensive
apps which are now currently operating way above their requested
memory.

This change fixes it by tuning the `execArgv` of the worker processes
to split the memory available between them, so the `node` runtime
tries to keep them below that threshold.

#### What problem is this solving?
Memory abuse by most memory-intensive apps.
e.g. [vtex.render-to-string](http://grafana.ingress.vtex.io/d/0FbY1kRWk/io-deployments?viewPanel=18&orgId=1&var-datasource=prometheus%20stores-2a&var-deployment_namespace=vendor-vtex&var-deployment_name=vtex-render-to-string-1-4-0-42b9058a3b29399)

#### How should this be manually tested?
Deploy runtime with this change in a cluster and make sure that apps
stay below their requested memory.

Note: I'm thinking about tuning the `max-old-space-size` that we set in the runtime
so it's larger than 75% of requested memory, as we can probably use more than
that since we run in containers not VMs.

#### Types of changes
